tests/integration_tests/instances.py

```diff
@@ -35,6 +35,17 @@ def __init__(self, cloud: 'IntegrationCloud', instance: BaseInstance,
     def destroy(self):
         self.instance.delete()
 
+    def restart(self):
+        """Restart this instance (via cloud mechanism) and wait for boot.
+
+        This wraps pycloudlib's `BaseInstance.restart` to pass
+        `raise_on_cloudinit_failure=False` to `BaseInstance.wait`, mirroring
+        our launch behaviour.
+        """
+        self.instance.restart(wait=False)
```

Comment:
Ahh nice in uaclient we just added 

instance.stop(wait=True)
instance.start(wait=False)
instance.wait(raise_on_cloudinit_failure=False).

I think your approach here is a little more succinct simpler and I *think* more correct.